src/StaticCaching/Cachers/FileCacher.php

```diff
@@ -167,7 +167,17 @@ public function getFilePath($url, $site = null)
         $urlParts = parse_url($url);
         $pathParts = pathinfo($urlParts['path']);
         $slug = $pathParts['basename'];
-        $query = $this->config('ignore_query_strings') ? '' : Arr::get($urlParts, 'query', '');
+        $query = Arr::get($urlParts, 'query', '');
+
+        if ($this->config('ignore_query_strings')) {
```

Comment:
When we revisit this PR, we'll need to spend some time getting it working again when `ignore_query_strings` is `true`, as it looks like our changes in #10701 broke this. 😄

Before #10701, this method ensured that only query parameters configured in the `allowed_query_strings` array (like `page`) were included in the filename.

However, it now it seems like the `$url` being passed into this method already has the query parameters stripped off.